lib/Access.php

```diff
@@ -1515,9 +1515,13 @@ public function getUserDnByUuid($uuid) {
 
 		$filter = $uuidAttr . '=' . $uuid;
 		$result = $this->searchUsers($filter, ['dn'], 2);
-		if (isset($result[0]['dn']) && \count($result) === 1) {
-			// we put the count into account to make sure that this is
-			// really unique
+
+		// we put the count into account to make sure that this is really unique
+		if (\count($result) > 1) {
+			throw new \LengthException($uuidAttr . " is specified as UUID attribute but has a value '{$uuid}' for multiple entries");
```

Comment:
We need to update the PHPDocs and clarify when this exception is thrown. Note that I personally don't like to use predefined exceptions unless we throw them under the same assumptions they're expected to be thrown, which isn't the case, so the clarification is particularly important.

Reply:
done, see above